### ISSUE TITLE
Bug: redirect() method does not work with non existing route

### ISSUE THREAD

**Issue by najdanovicivan** [codeigniter4/CodeIgniter4#4732]
**Describe the bug**

Calling redirect method which not existing route eg. `redirect('/some/non/existing/route)` will throw HTTPException::forInvalidRedirectRoute($route) which is not supposed to happen.

The documentation states:

```
If nothing is found, will treat
as a traditional redirect and pass the string in, letting
$response->redirect() determine the correct method and code.
```

So in such case the redirect should go directly to the passed route.

**CodeIgniter 4 version**
development

**Affected module(s)**
Common.pjp, RedirectResponse

**Context**
 - OS: [e.g. Windows 99]
 - Web server [e.g. Apache 1.2.3]
 - PHP version [e.g. 6.5.4]

**Comment by iRedds** [comment]
redirect()->to()

**Comment by paulbalandan** [comment]
IMO this is more of a documentation error rather than a bug in the code. Looking at the code if you pass a route string to `redirect()` it will invoke `RouteCollection::reverseRoute` which will check for a named route or a registered route. Clearly, the route string passed must be defined, otherwise an `HTTPException` will be thrown.

Have you tried using @iRedds ' suggestion?

**Comment by najdanovicivan** [comment]
I understand how it works but docs states differentlly. Either method or docs should be updated so that those are in sync

**Comment by michalsn** [comment]
The only problem is docblock. There is no doubt how things are working after reading the user guide.

The code was working that way from the beginning so the best we can do is to correct the docblock.

**Comment by michalsn** [comment]
Resolved via #4884
